src/probnum/linalg/solvers/stopping_criteria/_maxiter.py

```diff
@@ -0,0 +1,32 @@
+"""Stopping criterion based on a maximum number of iterations."""
+from typing import Optional
+
+import probnum  # pylint: disable="unused-import"
+
+from ._linear_solver_stopping_criterion import LinearSolverStoppingCriterion
+
+
+class MaxIterationsStopCrit(LinearSolverStoppingCriterion):
+    r"""Stop after a maximum number of iterations.
+
+    Stop when the solver has taken a maximum number of steps. If ``None`` is
+    specified, defaults to :math:`10n`, where :math:`n` is the dimension
+    of the solution to the linear system.
+
+    Parameters
+    ----------
+    maxiter :
+        Maximum number of steps the solver should take.
+    """
+
+    def __init__(self, maxiter: Optional[int] = None):
+        self.maxiter = maxiter
+
+    def __call__(
+        self, solver_state: "probnum.linalg.solvers.ProbabilisticLinearSolverState"
+    ) -> bool:
+
+        if self.maxiter is None:
+            return solver_state.step >= solver_state.problem.A.shape[0] * 10
+        else:
+            return solver_state.step >= self.maxiter
```

Comment:
```suggestion
        if self.maxiter is None:
            self.maxiter = solver_state.problem.A.shape[0] * 10

        return solver_state.step >= self.maxiter
```

This would also increase test coverage :)

Reply:
On a related note: Do we actually want the `* 10` in there yet? I guess the algorithm should terminate after `n` steps in the absence of noise, right?

Reply:
Not in finite arithmetic. This is also the default chosen by `scipy.sparse.linalg.cg`: ([code snippet](https://github.com/scipy/scipy/blob/9629282c2aa9e66b898357577aeec6bcb7e6f3c6/scipy/sparse/linalg/isolve/iterative.py#L294)).